src/main/java/com/conveyal/datatools/manager/models/transform/AddCustomFileFromStringTransformation.java

```diff
@@ -0,0 +1,24 @@
+package com.conveyal.datatools.manager.models.transform;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+
+public class AddCustomFileFromStringTransformation extends StringTransformation {
+
+    // Additional create method required to ensure transformation type is AddCustomFile in tests.
+    // Otherwise, we'd use the StringTransformation#create which doesn't differentiate types and hence
+    // would fail table name tests.
+    public static AddCustomFileFromStringTransformation create(String csvData, String table) {
+        AddCustomFileFromStringTransformation transformation = new AddCustomFileFromStringTransformation();
+        transformation.csvData = csvData;
+        transformation.table = table;
```

Comment:
This can be for a separate PR: consider reporting the number of rows inserted for this transformation.

Reply:
Added it into this PR, the front end has an update to support this as well: https://github.com/ibi-group/datatools-server/pull/536/commits/ec87de24d086d86115fc5f95b7cd539c74eb3d3b